apps/rush-lib/src/scripts/install-run.ts

```diff
@@ -1,4 +1,104 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
-console.log('Invoked install-run');
+// This file was generated by a tool. Modifying this file will produce unexpected behavior.
+//
+// This script is used to install and invoke a tool from a CI definition.
+
+import * as childProcess from 'child_process';
+
+import {
+  installAndRun,
+  IPackageSpecifier,
+  getNpmPath,
+  runWithErrorPrinting
+} from './install-run-common';
+
+/**
+ * Parse a package specifier (in the form of name\@version) into name and version parts.
+ */
+function parsePackageSpecifier(rawPackageSpecifier: string): IPackageSpecifier {
+  rawPackageSpecifier = (rawPackageSpecifier || '').trim();
+  const separatorIndex: number = rawPackageSpecifier.lastIndexOf('@');
+
+  let name: string;
+  let version: string | undefined = undefined;
+  if (separatorIndex === 0) {
+    // The specifier starts with a scope and doesn't have a version specified
+    name = rawPackageSpecifier;
+  } else if (separatorIndex === -1) {
+    // The specifier doesn't have a version
+    name = rawPackageSpecifier;
+  } else {
+    name = rawPackageSpecifier.substring(0, separatorIndex);
+    version = rawPackageSpecifier.substring(separatorIndex + 1);
+  }
+
+  if (!name) {
+    throw new Error(`Invalid package specifier: ${rawPackageSpecifier}`);
+  }
+
+  return { name, version };
+}
+/**
+ * Resolve a package specifier to a static version
+ */
+function resolvePackageVersion({ name, version }: IPackageSpecifier): string {
+  if (!version) {
+    version = '*'; // If no version is specified, use the latest version
+  }
+
+  if (version.match(/[\*\^\~]/ig)) {
```

Comment:
>match [](start = 14, length = 5)

Might be easier to match known good things, rather than try to catch all possible bad things.

`^[a-z0-9\-\+]+$`